include/mruby/value.h

```diff
@@ -435,10 +435,20 @@ mrb_ro_data_p(const char *p)
 #elif defined(__APPLE__)
 #define MRB_LINK_TIME_RO_DATA_P
 #include <mach-o/getsect.h>
+#include <crt_externs.h> // for _NSGetMachExecuteHeader
 static inline mrb_bool
 mrb_ro_data_p(const char *p)
 {
-  return (char*)get_etext() < p && p < (char*)get_edata();
+#ifdef __LP64__
+  struct mach_header_64 *mhp;
+#else
+  struct mach_header *mhp;
+#endif
+  mhp = _NSGetMachExecuteHeader();
+  unsigned long textsize, datasize;
+  char *text = (char*)getsegmentdata(mhp, SEG_TEXT, &textsize);
+  char *data = (char*)getsegmentdata(mhp, SEG_DATA, &datasize);
+  return text + textsize < p && p < data + datasize;
```

Comment:
I don't have macOS so I can't check, but I have two concerns:

  - Does macOS guarantee that the "text" segment and the "data" segment will be contiguous in the future?
  - Wouldn't it be enough to check the extent of the "data" segment?

    ```c
    return (unsigned long)(p - data) < datasize;
    ```